### ISSUE TITLE
All-Zero Separations

### ISSUE THREAD

**Issue by thesmith1** [bytedance/music_source_separation#25]
Hi,

first of all, thank you for open-sourcing the code!

I've cloned the repository and downloaded the pretrained models with the `./separate_scripts/download_checkpoints.sh` script.

By looking at the weights of the model (`resunet143_ismir2021_vocals_8.9dB_350k_steps.pth`), I noticed that the running_var of the batchnorm layers in the first decoding blocks has very high mean values, even reaching 2*10^4: is this expected?

The reason I'm asking is the following.
I'm running inference using this script:
```
#!/bin/bash

WORKSPACE=<my_workspace_path>

echo "WORKSPACE=${WORKSPACE}"
export PYTHONPATH=./

# Users can modify the following config file.
TRAIN_CONFIG_YAML="scripts/4_train/musdb18/configs/vocals-accompaniment,resunet_ismir2021.yaml"

CHECKPOINT_PATH="<my_path_to_checkpoints>/resunet143_ismir2021_vocals_8.9dB_350k_steps.pth"

IN_FILE="<my_path_to_demo_songs>/PanicStation.wav"
OUT_FILE="<my_path_to_results>/PS_separations.wav"

# Inference
python3 bytesep/inference.py \
    --config_yaml=$TRAIN_CONFIG_YAML \
    --checkpoint_path=$CHECKPOINT_PATH \
    --audio_path=$IN_FILE \
    --output_path=$OUT_FILE
```

The output contains all zeros, in all channels.

By looking at the intermediate feature maps, I noticed that their values increase while going deeper into the architecture. This happens gradually, until the variable `x_center` at [this](https://github.com/bytedance/music_source_separation/blob/0a088e1fc852a15d7e558a7e203888de0577dfb1/bytesep/models/resunet_ismir2021.py#L503) line contains values such as 10^36.
At that point, I get infinity, followed by NaNs, which eventually result in all-zeros output.

I'm not sure this is related to the high values in the running_var, but the deeper architecture is the only one that causes this trouble (the simple UNet does not).

Does this happen to some of you as well?

**Comment by qiuqiangkong** [comment]
Hi, thanks! Did you get correct separation result for resources/vocals_accompaniment_10s.mp3

**Comment by thesmith1** [comment]
Hi, thanks for your reply!

Yes, I tried also with that sample, the output is still all-zero.

Some more info below.
I've tested different architectures and noticed that the output is zero only with `scripts/4_train/musdb18/configs/vocals-accompaniment,resunet_ismir2021.yaml` (PT model: `downloaded_checkpoints/resunet143_ismir2021_vocals_8.9dB_350k_steps.pth`).
I've tried `scripts/4_train/musdb18/configs/vocals-accompaniment,resunet_subbandtime.yaml` (PT model: `downloaded_checkpoints/resunet143_subbtandtime_vocals_8.8dB_350k_steps.pth`) (with the same script as above), and the separation is correct.

I also re-trained from scratch the network in this config file (`scripts/4_train/musdb18/configs/vocals-accompaniment,unet.yaml`) and it works as expected.

Finally, I re-trained the architecture that does not work (`vocals-accompaniment,resunet_ismir2021.yaml`): although everything goes well during training (based on the evaluation scores that are printed), during inference I still get all-zero outputs.

**Comment by qiuqiangkong** [comment]
It is suggested to use subband_time versions because it is 4x faster and the performance is close to the ISMIR version. One possible reason for all-zero may caused by not installing InplaceABN correctly: https://github.com/mapillary/inplace_abn
